v2/spanner-common/src/test/java/com/google/cloud/teleport/v2/spanner/migrations/avro/GenericRecordTypeConvertorTest.java

```diff
@@ -0,0 +1,603 @@
+import com.google.cloud.Date;
+import com.google.cloud.Timestamp;
+import com.google.cloud.spanner.Dialect;
+import com.google.cloud.spanner.Value;
+import com.google.cloud.teleport.v2.spanner.ddl.Ddl;
+import com.google.cloud.teleport.v2.spanner.migrations.exceptions.AvroTypeConvertorException;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.ISchemaMapper;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.IdentityMapper;
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import com.google.cloud.teleport.v2.utils.SchemaUtils;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.nio.ByteBuffer;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.junit.Test;
+
+public class GenericRecordTypeConvertorTest {
+
+  @Test
+  public void testAvroFieldToBoolean() {
+    Boolean inputValue = true;
+    Boolean result =
+        GenericRecordTypeConvertor.avroFieldToBoolean(
+            inputValue, SchemaBuilder.builder().booleanType());
+    assertEquals("Test true boolean input", inputValue, result);
+
+    inputValue = false;
+    result =
+        GenericRecordTypeConvertor.avroFieldToBoolean(
+            inputValue, SchemaBuilder.builder().booleanType());
+    assertEquals("Test false boolean input", inputValue, result);
+
+    result =
+        GenericRecordTypeConvertor.avroFieldToBoolean("true", SchemaBuilder.builder().stringType());
+    assertEquals("Test string input", true, result);
+
+    result = GenericRecordTypeConvertor.avroFieldToBoolean(4, SchemaBuilder.builder().intType());
+    assertEquals("Test int input", false, result);
+
+    result = GenericRecordTypeConvertor.avroFieldToBoolean(1L, SchemaBuilder.builder().longType());
+    assertEquals("Test long input", false, result);
+  }
+
+  @Test
+  public void testAvroFieldToBoolean_NullInput() {
+    Boolean result =
+        GenericRecordTypeConvertor.avroFieldToBoolean(null, SchemaBuilder.builder().stringType());
+    assertNull(result);
+  }
+
+  @Test
+  public void testAvroFieldToLong_ValidConversion() {
+    Long inputValue = 10L;
+    Long result =
+        GenericRecordTypeConvertor.avroFieldToLong(inputValue, SchemaBuilder.builder().longType());
+    assertEquals(inputValue, result);
+
+    // Test integer values as well.
+    Integer intValue = 10;
+    result =
+        GenericRecordTypeConvertor.avroFieldToLong(inputValue, SchemaBuilder.builder().intType());
+    assertEquals("Test int input", Long.valueOf(intValue), result);
+
+    // Test string values as well.
+    result = GenericRecordTypeConvertor.avroFieldToLong("1536", SchemaBuilder.builder().intType());
+    assertEquals("Test string input", (Long) 1536L, result);
```

Comment:
Just because we got burned by a bug on long conversion recently, it would be a good idea if we can use MAX and MIN values here instead of general values. E.g Use `LONG_MAX` and `LONG_MIN` here. 
Applicable to other places.